### PR TITLE
[pkp-lib] #2947 NMI parent with no displayable children fix

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

**Comment by NateWr** [comment]
I may be nitpicking, but I'd prefer if this logic wasn't in the template. I try to keep the templates as simple as possible, since less-experienced developers interact with them. 

Can we calculate if a parent item has displayable children in `NavigationMenuService::getMenuTree()` and attach it to the parent item, so it's just a `boolean` that we can check in the templates?

**Comment by defstat** [comment]
@NateWr I have made some work on the Caching management of the NavigationMenus (see #2951). There I have changed the order of the call for `getDisplayStatus` function, and it's more suitable to add the suggested functionality there. So we can do one of the following: 
- wait for the caching feature to merge
- change the functionality of the `getMenuTree` to match both branches somehow, which I think is the best option.

**Comment by defstat** [comment]
@NateWr I ended up adding that here.

File: templates/frontend/components/navigationMenu.tpl [hunk 1]
```diff
@@ -23,17 +23,17 @@
...
+                    <ul>
+					{foreach key=childField item=childNavigationMenuItemAssignment from=$navigationMenuItemAssignment->children}
+						{if $childNavigationMenuItemAssignment->navigationMenuItem->getIsDisplayed()}
+							<li class="{$liClass|escape}">
+								<a href="{$childNavigationMenuItemAssignment->navigationMenuItem->getUrl()}">
+									{$childNavigationMenuItemAssignment->navigationMenuItem->getLocalizedTitle()}
+								</a>
+							</li>
+						{/if}
+					{/foreach}
```
Comment (NateWr) [inline templates/frontend/components/navigationMenu.tpl]: Some indentation issues with the `{foreach}` and the `<ul>` above.

Comment (defstat) [reply templates/frontend/components/navigationMenu.tpl]: Fixed

**Comment by NateWr** [comment]
This looks like it's ready to go @defstat. It just needs a rebase against the current `master` branch.

**Pull request closed without merging**